Fixes #2001: Added bottom padding to improve visual break

rjayroso opened the pull request:
## Issue This PR Addresses
Fixes #2001

## Type of Change
- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description
I've thought about how to improve the visual break between posts. I've looked at how Instagram separates their infinite timeline of posts and it turns out that they don't (unlike tumblr). I think that the easiest solution would just to simply give it more space, see below:

https://user-images.githubusercontent.com/48869737/114341111-fe278f00-9b26-11eb-857c-7494dfe24f7f.mp4

From my reasoning, the next post's title looks like a subtitle for the current post is because the spacing is basically equal and the user struggles to identify who the title belongs to. By adding more space between, it'll be easier to see the title belongs to the post below rather than the previous post above. 

Let me know if we want a more drastic design change to improve the visual break or if the gap should be shortened or made larger.

## How to Test
1. View the file change, ask the following:
    - should I change the unit of measurement? (`px` instead of `rem`?)
2. View deployment, ask the following:
    - is the extra space enough? too little?
    - should this warrant a more drastic design change?

## Checklist
- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

PedroFonsecaDEV commented:
@rjayroso @humphd 

I think that increasing the space between the posts is not enough. We need to add something else. 
Follow my suggestions:
# 1
<img width="1001" alt="Screen Shot 2021-04-12 at 5 15 51 AM" src="https://user-images.githubusercontent.com/60857409/114371188-3f7f6500-9b4e-11eb-964e-1d316f0df374.png">


# 2 (I'm using the wrong colors here).

**gif:**
<br/>
![2021-04-12 05 30 21](https://user-images.githubusercontent.com/60857409/114373194-5a52d900-9b50-11eb-9944-f245fdada84b.gif)

<br/>
<br/>

<br/>
<br/>

**img:**

<img width="1095" alt="Screen Shot 2021-04-12 at 5 30 52 AM" src="https://user-images.githubusercontent.com/60857409/114373196-5a52d900-9b50-11eb-9219-39d6ea771cde.png">

PedroFonsecaDEV reviewed (changes requested):
Please check the comment above.

rjayroso commented:
@PedroFonsecaDEV  I may have encountered a bug that was giving me trouble, though it may be intentional. That aside, do we still want to put the avatar at the bottom? I think when it is scaled down it will be hard to see and just be noise. I added the line and the "Written by: author" though and copied the colours and border width from the author container on the side, let me know what you think. 

![image](https://user-images.githubusercontent.com/48869737/114984608-e8d09e80-9e5f-11eb-9262-9fdf706935a9.png)

https://user-images.githubusercontent.com/48869737/114985364-b96e6180-9e60-11eb-8c4d-6861d36d5889.mp4

---

**As for the aforementioned bug, there is an invisible image at the end of all medium posts.**

```css
.telescope-post-content img[src*='medium.com/_/stat'] {
  width: 1px;
  height: 1px;
}
```
![image](https://user-images.githubusercontent.com/48869737/114985808-3d284e00-9e61-11eb-8f6a-37d4bed67262.png)

If we put a line at the bottom, the gap will appear larger (specifically to medium posts).
I am unsure if I should remove it since I don't know if it serves a purpose elsewhere, thoughts?

PedroFonsecaDEV commented on src/web/src/components/Posts/Post.tsx:
```diff
@@ -172,6 +172,12 @@ const useStyles = makeStyles((theme: Theme) =>
       },
     },
+    contentFooter: {
+      borderBottom: '2.5px solid #707070',
```
I think we can go with `1.5px`

PedroFonsecaDEV commented on src/web/src/components/Posts/Post.tsx:
```diff
@@ -302,6 +308,14 @@ const PostComponent = ({ postUrl }: Props) => {
       </div>
+      <div className={classes.contentFooter}>
+        <p>
+          Written by:{' '}
```
Let's add the <TelescopeAvatar> as well.

chrispinkney replied:
I'm not sure how I feel about `Written by`, we already show the user on the right, why add it again at the bottom? I do think the line is nice though.

What do others think?

PedroFonsecaDEV reviewed (changes requested):
I made some comments.
Regarding the medium ghost image, we can deal with it in a follow-up. Let's keep the scope of this PR.

humphd commented:
What if we try only doing the bar with no extra info about the user or post for now, and iterate on that?  That seems to be the general feedback I'm hearing on our call today.

huynguyez commented:
@rjayroso @humphd I'll stick with the idea I mentioned earlier in the call:

Keep the header and add a line like the picture below (+ some margin to the upper post), this not only help 2 separate posts more visible but also we can remove the current author info on right side. That place, instead of author info, we can use that space (right) for the share feature in future, or a 'monthly/yearly timeline' (help user browsing old posts by that specific month/year when click on).

![Screen Shot 2021-04-16 at 4 51 53 PM](https://user-images.githubusercontent.com/10421864/115082877-98475880-9ed4-11eb-9866-8b52fe4ee948.jpg)

rjayroso commented:
@PedroFonsecaDEV @huynguyez @chrispinkney, what do you think of the latest [deployment](https://telescope-67bgs5iao-humphd.vercel.app/)? I think for the whole removing the sidebar thing should be done in another issue, lets just get the line and spacing for the visual break. 

Anyways, should the line be close to the top post, in the middle, or as Huy suggested - closer to the bottom? Also: any comments on making a larger/smaller gap, darker/lighter, thinner/bolder line etc?

chrispinkney reviewed (changes requested):
![image](https://user-images.githubusercontent.com/7242003/115095326-dea9b100-9eee-11eb-9bf2-5eb20b5e9e11.png)

Looks great! Just needs some more `margin-top` I think, since the gap between the next post and the end of the current post are noticeably different. If others are fine with it then I don't mind giving an approval!

rjayroso commented:
> ![image](https://user-images.githubusercontent.com/7242003/115095326-dea9b100-9eee-11eb-9bf2-5eb20b5e9e11.png)
> 
> Looks great! Just needs some more `margin-top` I think, since the gap between the next post and the end of the current post are noticeably different. If others are fine with it then I don't mind giving an approval!
> 
> I'm not sure what a darker/bolder line would look like but I'm curious to try it. Feel free to suggest some css changes I can apply to the deployment via the browser!

@chrispinkney done n done! rebased and squashed too

PedroFonsecaDEV reviewed (approved):
LGTM.
